resources/guidelines/code/backward-compatibility.md

```diff
@@ -0,0 +1,751 @@
+        <td></td>
+    </tr>
+    <tr>
+        <td>Remove a public property, constant or method.</td>
+        <td>🔴 NO</td>
+        <td>Annotate it as deprecated and remove it in the next major release.</td>
+    </tr>
+    <tr>
+        <td>Remove a protected property, constant or method.</td>
+        <td>🔴 NO</td>
+        <td>Annotate it as deprecated and remove it in the next major release.</td>
+    </tr>
+    <tr>
+        <td>Remove a private property, constant, or method.</td>
+        <td>✅ YES</td>
+        <td></td>
+    </tr>
+</table>
+
+### Storefront
+#### TWIG Templates
+<table>
+	<tr>
+		<th width=260>Use Case</th>
+		<th width=120>Allowed?</th>
+		<th>Notes / Alternatives</th>
+	</tr>
+	<tr>
+		<td>Removing TWIG blocks.</td>
+		<td>🔴 NO</td>
+		<td>Use the deprecation workflow.<br><br>Code Example: [Deprecate TWIG block](#deprecate-twig-block)</td>
+	</tr>
+	<tr>
+		<td>Renaming TWIG blocks.</td>
+		<td>🔴 NO</td>
+		<td>Use the deprecation workflow. Create a new surrounding block with the new name and deprecate the old one. All variables which are defined in the scope of the old block, must be moved to the new surrounding block scope.<br><br>Code Example: [Rename TWIG block](#rename-twig-block)</td>
+	</tr>
+	<tr>
+		<td>Moving TWIG blocks within the same file.</td>
+		<td>⚪ PARTIAL</td>
+		<td>Only within the same scope / parent block.</td>
+	</tr>
+	<tr>
+		<td>Removing TWIG variables.</td>
+		<td>🔴 NO</td>
+		<td></td>
+	</tr>
+	<tr>
+		<td>Renaming TWIG variables.</td>
+		<td>🔴 NO</td>
+		<td>Create a new variable within the same scope and deprecate the old one.</td>
+	</tr>
+	<tr>
+		<td>Changing the value of TWIG variables</td>
+		<td>⚪ PARTIAL</td>
+		<td>The data type has to stay the same. Otherwise use the deprecation workflow.</td>
+	</tr>
+	<tr>
+		<td>Moving TWIG variable definitions to other TWIG blocks.</td>
+		<td>⚪ PARTIAL</td>
+		<td>Only when they are being moved higher up in the block scope.</td>
+	</tr>
+	<tr>
+		<td>Adding TWIG blocks which affect the scope of variable definitions.</td>
+		<td>🔴 NO</td>
+		<td></td>
+	</tr>
+	<tr>
+		<td>Moving template files to other directories.</td>
+		<td>🔴 NO</td>
+		<td></td>
+	</tr>
+</table>
+
+#### HTML
+<table>
+	<tr>
+		<th width=260>Use Case</th>
+		<th width=120>Allowed?</th>
+		<th>Notes / Alternatives</th>
+	</tr>
+	<tr>
+		<td>Removing HTML sections.</td>
+		<td>🔴 NO</td>
+		<td></td>
+	</tr>
+	<tr>
+		<td>Moving HTML sections within the same file.</td>
+		<td>⚪ PARTIAL</td>
+		<td>Only within the same TWIG Block.</td>
+	</tr>
+	<tr>
+		<td>Renaming of removing CSS selectors.</td>
+		<td>🔴 NO</td>
+		<td>Use the deprecation workflow.<br><br>Code Example: [Deprecate CSS selectors](#deprecate-css-selectors)</td>
+	</tr>
+</table>
+
+#### JavaScript
+<table>
+	<tr>
+		<th width=260>Use Case</th>
+		<th width=120>Allowed?</th>
+		<th>Notes / Alternatives</th>
+	</tr>
+	<tr>
+		<td>Renaming or removing JS services.</td>
+		<td>🔴 NO</td>
+		<td></td>
```

Comment:
Doesn't the deprecation workflow apply here as well? And the same for "Renaming or removing of JS plugins"?

Reply:
Yes there was an example in the past, not sure why this wasn't in the GitLab version. Will try to clear this up.